contracts/nft/nft-pre-designed-collection.ral

```diff
@@ -11,7 +11,10 @@ Contract NFTPreDesignedCollection(
     @using(preapprovedAssets = true, updateFields = true)
     pub fn mint() -> (ByteVec) {
         let minter = callerAddress!()
-        let initialImmState = EnumerableNFT.encodeImmFields!(selfContractId!(), totalSupply)
+        let initialImmState = EnumerableNFT.encodeImmFields!(
+            NFTPreDesignedCollection(selfContractId!()),
```

Comment:
It seems we should introduce a new built-in function `selfContract!()` for this line

Reply:
Good idea! I can add that.